nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/trash/LifeCycleTrashService.java

```diff
@@ -0,0 +1,184 @@
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Kevin Leturc <kleturc@nuxeo.com>
+ */
+package org.nuxeo.ecm.core.trash;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentRef;
+import org.nuxeo.ecm.core.api.DocumentSecurityException;
+import org.nuxeo.ecm.core.api.LifeCycleConstants;
+import org.nuxeo.ecm.core.api.security.SecurityConstants;
+
+/**
+ * @deprecated since 10.1, use {@link PropertyTrashService} instead.
+ */
+public class LifeCycleTrashService extends AbstractTrashService {
+
+    private static final Log log = LogFactory.getLog(LifeCycleTrashService.class);
+
+    @Override
+    public boolean isTrashed(CoreSession session, DocumentRef docRef) {
+        return LifeCycleConstants.DELETED_STATE.equals(session.getCurrentLifeCycleState(docRef));
+    }
+
+    @Override
+    public void trashDocuments(List<DocumentModel> docs) {
+        if (docs.isEmpty()) {
+            return;
+        }
+        CoreSession session = docs.get(0).getCoreSession();
+        for (DocumentModel doc : docs) {
+            // add context data for backward compatibility mechanism
+            doc.putContextData(IS_ALREADY_CALLED, Boolean.TRUE);
+            DocumentRef docRef = doc.getRef();
+            if (session.getAllowedStateTransitions(docRef).contains(LifeCycleConstants.DELETE_TRANSITION)
+                    && !doc.isProxy()) {
+                if (!session.canRemoveDocument(docRef)) {
+                    throw new DocumentSecurityException("User " + session.getPrincipal().getName()
+                            + " does not have the permission to remove the document " + doc.getId() + " ("
+                            + doc.getPath() + ")");
+                }
+                trashDocument(session, doc);
+            } else if (session.isTrashed(docRef)) {
+                log.warn("Document " + doc.getId() + " of type " + doc.getType()
+                        + " is already in the trash, nothing to do");
+                return;
+            } else {
+                log.warn("Document " + doc.getId() + " of type " + doc.getType() + " in state "
+                        + doc.getCurrentLifeCycleState() + " does not support transition "
+                        + LifeCycleConstants.DELETE_TRANSITION + ", it will be deleted immediately");
+                session.removeDocument(docRef);
+            }
+        }
+        session.save();
+    }
+
+    protected void trashDocument(CoreSession session, DocumentModel doc) {
+        if (doc.getParentRef() == null) {
+            // handle placeless document
+            session.removeDocument(doc.getRef());
+        } else {
+            if (!Boolean.parseBoolean(String.valueOf(doc.getContextData(DISABLE_TRASH_RENAMING)))) {
```

Comment:
You can compare directly, no need go through a String: `if (!Boolean.TRUE.equals(doc.getContextData(DISABLE_TRASH_RENAMING)))`